Use ui5-test-runner instead of deprecated karma

akudev opened the pull request:
WIP

akudev committed:
Use ui5-test-starter instead of deprecated karma

Fixes #30

akudev commented:
Open:
- documentation needs to be adapted
- `--port 8081` doesn't seem to do anything reasonable

arnaudbuchholz-sap reviewed (approved):
I confirm that using `--port 8081` is useless if you don't intend to use it. `ui5-test-runner` would allocated a random free port to receive updates from the tests.

For documentation, feel free to push a PR to `ui5-test-runner`

akudev committed:
Update README

akudev committed:
Update code

akudev committed:
Vast updates to step-by-step.md

Everything changed and covered, but the goal is now to replace the main
branch with this,so some further effort is needed to:
- move some more README content to step-by-step.md
- remove references to main
- get any main branch content or updates over

akudev committed:
Final tweaks

akudev committed:
More tweaks

akudev committed:
Code tweaks

akudev commented:
Hi @arnaudbuchholz-sap and @petermuessig, I have now also adapted all documentation. (Which led to some further improvements/fixes in the code). Please have another look, as we want to
1. replace the `main` branch of this repo with this content later to make testing an integral part (this will require some syncing of content updates back and forth, so I will do it in a separate step)
2. adapt the outdated Easy-UI5 templates according to this code, especially the test code and config

ArnaudBuchholz commented on package.json:
```diff
@@ -9,14 +9,15 @@
-    "karma-ci": "karma start karma-ci.conf.js",
-    "karma-ci-cov": "karma start karma-ci-cov.conf.js",
-    "test": "npm run lint && npm run karma-ci-cov"
+    "test-runner": "ui5-test-runner --port 8081 --url http://localhost:8080/test/testsuite.qunit.html",
+    "test-runner-coverage": "ui5-test-runner --port 8081 --url http://localhost:8080/test/testsuite.qunit.html --coverage -ccb 60 -ccf 100 -ccl 85 -ccs 85",
+    "test-ui5": "start-server-and-test start-coverage http://localhost:8080 test-runner-coverage",
```
2 things :
* The `--port` option is useless unless you want to monitor the progress. It may be confusing for people looking at it
* `ui5-test-runner` now offers the `--start` option which allows you to eliminate `start-server-and-test` 😄

akudev replied:
Done, thanks!

ArnaudBuchholz reviewed (changes requested):
You can improve a bit the use of `ui5-test-runner`

akudev committed:
Make the optional test-runner monitoring port less prominent

akudev committed:
Use the new --start parameter of ui5-test-runner
